src/world/footpath.c

```diff
@@ -1506,7 +1506,30 @@ void sub_6A759F()
  */
 static void footpath_unown(int x, int y, rct_map_element *pathElement)
 {
-	map_buy_land_rights(x, y, x, y, 6, 1);
+	rct_map_element *surfaceElement = map_get_surface_element_at(x >> 5, y >> 5);
+	int ownership = surfaceElement->properties.surface.ownership;
+	if (ownership == 0)
+		return;
+	int type = 0, xx = x, yy = y, cmd = GAME_COMMAND_SET_LAND_OWNERSHIP, flags = GAME_COMMAND_FLAG_APPLY;
+
+	//if footpath is higher than 2 units or lower than land, do not own land.
+ 	//doing so will confuse outside guests.
+ 	int z = pathElement->base_height;
+ 	if (z != surfaceElement->base_height) {
+ 		if (z > surfaceElement->base_height + 2 || z < surfaceElement->base_height) {
+			type = 0;
+			game_command_set_land_ownership(&xx, &flags, &yy, &type, &cmd, &xx, &yy);
+			return;
+  		}
+  	}
+	if (ownership == OWNERSHIP_OWNED) {
+		type = 1;
```

Comment:
In https://github.com/OpenRCT2/OpenRCT2/pull/4825/commits/9033673fd1d0b398b5cb2a2797af6dad83147378 I left some comments on the assumed types, 

https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/world/park.c#L950 there is also the link to the function that takes the type value.. 
doing a quick skim of the code in the line above, i'm guessing the type values are interpreted as the following
0 - buy land ownership if it's for sale
1 - removes land ownership and rights. (tile may still be available for purchase after this, unsure)
2 - buys rights if it's for sale.
3 - removes rights from tile. (may still be available for purchase after this, unsure)
4 - marks land for sale.
5 - marks rights for sale
6 and above - unsure exactly just yet

i could run some tests against them later to verify the exact usage of the setting, but for now would you possibly be able to help suggest a useful constant for them? or at least a format or couple examples i could follow? there are already enums for "OWNERSHIP_AVAILABLE" etc, but those don't reflect the same numbers that are used for the settings above